### PR TITLE
Fix tkn p start --showlog by handling informer ResourceEventHandler

### PR DESCRIPTION
**Pull request by pratap0007**
Fix tkn p start --showlog by handling informer ResourceEventHandler
    
tkn p start --showlog command was failing as informer ResourceEventHandler
was calling UpdateFunc after closing channel so it was crashing and showing
error `sending data on closed channel`
    
so this patch handles informer ResourceEventHandler by synchronization and checks
if channel is closed then return and does nothing

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```release-note
This will fix ` tkn p start --showlog` command  by handling informer ResourceEventHandler to show the logs
```

### CODE REVIEW

File: pkg/pipelinerun/tracker.go [hunk 1]
```diff
@@ -107,8 +106,36 @@ func (t *Tracker) Monitor(allowed []string) <-chan []taskrunpkg.Run {
...
+				mu.Lock()
+				defer mu.Unlock()
+				select {
+				case <-stopC:
+					return
+				default:
+					eventHandler(newObj)
+				}
+			},
+			DeleteFunc: func(obj interface{}) {
```
Comment (vinamra28) [inline pkg/pipelinerun/tracker.go:129]: can you please add comments so that it can be clear to the person who visits this code next time

**Review by vinamra28 — COMMENTED** [review]
/lgtm

**Comment by piyush-garg** [comment]
@pratap0007 can you update the commit and PR description with the cause and solution for the issue?

**Comment by piyush-garg** [comment]
/lgtm
cc @vdemeester

**Pull request merged**